packages/patternfly-4/react-core/src/components/Progress/Progress.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Progress/styles.css';
+import { css, getModifier } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import ProgressContainer, { ProgressMeasureLocation, ProgressStatus } from './ProgressContainer';
+
+export { ProgressMeasureLocation, ProgressStatus } from './ProgressContainer';
+
+export const ProgressVariant = {
+  sm: 'sm',
+  md: 'md',
+  lg: 'lg'
+};
+
+const propTypes = {
+  /** Classname for progress component. */
+  className: PropTypes.string,
+  /** Size variant of progress. */
+  variant: PropTypes.oneOf(Object.keys(ProgressVariant)),
+  /** Where the measure percent will be located. */
+  measureLocation: PropTypes.oneOf(Object.keys(ProgressMeasureLocation)),
+  /** Progress status type. */
+  status: PropTypes.oneOf(Object.keys(ProgressStatus)),
+  /** Title above progress. */
+  title: PropTypes.string,
+  /** Actual value of progress. */
+  value: PropTypes.number,
+  /** DOM id for progress component. */
+  id: PropTypes.string,
+  /** Minimal value of progress. */
+  valueMin: PropTypes.number,
+  /** Maximum value of progress. */
+  valueMax: PropTypes.number,
+  /** Dynamic description of progress. */
+  valueText: PropTypes.string
+};
+
+const defaultProps = {
+  className: '',
+  measureLocation: ProgressMeasureLocation.top,
+  status: ProgressStatus.info,
+  id: `progress-${new Date().getTime()}${Math.random()
+    .toString(36)
+    .slice(2)}`,
+  title: '',
+  valueMin: 0,
+  valueMax: 100,
+  variant: null,
+  get value() {
+    return defaultProps.valueMin || 0;
```

Comment:
This will always evaluate to 0. No need for the get here.

Reply:
Well if consumer sets min to `40` let's say but no value the value should stay at `40` not `0`.

Reply:
A consumer or react would need to mutate this default props object for that. To achieve that functionality it will need to be done in the render function.